bundles/binding/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/connection/ConnectionManager.java

```diff
@@ -0,0 +1,286 @@
+	 */
+	public void add(Connection conn) {
+		this.connectionsById.put(conn.getSets().getId().toUpperCase(), conn);
+	}
+	
+	/**
+	 * Closes and removes the given connection.
+	 * 
+	 * @param id the connection's unique identifier
+	 * @return true on success
+	 */
+	public boolean disconnectAndRemove(String id) {
+		Connection conn = this.connectionsById.get(id.toUpperCase());
+		if (conn == null)
+			return false;
+		conn.disconnect();
+		this.connectionsById.remove(id);		
+		return true;
+	}
+
+	/**
+	 * Find a connection by its unique identifier.
+	 * 
+	 * @param id the connection to search for
+	 * @return the found connection or null
+	 */
+	public Connection get(String id) {
+		return this.connectionsById.get(id.toUpperCase());
+	}
+
+	/** Notifies that the binding has started. */
+	public void activate() {
+		try {
+			this.selector = Selector.open();
+			this.thread = new Thread(this);
+			this.thread.start();			
+		} catch (IOException e) {
+			LcnLogger.error("Unable to open the Selector!");
+		}	
+	}
+
+	/** Notifies that the binding has stopped. */	
+	public void deactivate() {
+		try {
+			this.threadTreminate = true;
+			this.selector.close();
+			this.thread.join();
+		} catch (InterruptedException ex) {
+		} catch (IOException e) {
+			LcnLogger.error("Unable to close the Selector!");
+		}
+	}
+	
+	/** Tells all connections to flush their queued data. */
+	public void flush() {
+		for (Connection conn : this.connectionsById.values())
+			conn.flush();
+	}
+
+	/**
+	 * Updates the managed connections.
+	 * Obsolete connections are closed and removed, while new and updated ones are (re-)established.
+	 * Finally all connections are told to update themselves. 
+	 */	
+	public void update() {
+		// Close and remove changed and no longer existing connections
+		HashSet<String> removeIds = new HashSet<String>(); 
+		for (Connection conn : this.connectionsById.values()) {
+			ConnectionSettings sets = this.callback.getAllSets().get(conn.getSets().getId().toUpperCase());
+			if (sets != null) {
+				if (!sets.equals(conn.getSets()))
+					removeIds.add(conn.getSets().getId());
+			}
+			else
+				removeIds.add(conn.getSets().getId());
+		}
+		for (String id : removeIds)
+			this.disconnectAndRemove(id);
+		// Add new connections
+		for (ConnectionSettings sets : this.callback.getAllSets().values()) {
+			if (!this.connectionsById.containsKey(sets.getId().toUpperCase())) {
+				Connection conn = new Connection(sets, this);
+				this.add(conn);
+				conn.beginConnect();
+			}
+		}
+		// Finally update now existing connections
+		for (Connection conn : this.connectionsById.values())
+			conn.update();
+	}
+	
+	/** {@inheritDoc} */
+	@Override
+	public Selector getSelector() {
+		return this.selector;
+	}
+	
+	/** {@inheritDoc} */
+	@Override
+	public void updateItems(Connection conn) {
+		this.callback.updateItems(conn);
+	}
+	
+	/** {@inheritDoc} */
+	@Override
+	public void processInput(String input, Connection conn) {
```

Comment:
better rename to onInputReceived (or alike) in order to express the callback/event nature of this method